Assets/SplineMesh/Scripts/Editor/SplineEditor.cs

```diff
@@ -179,8 +215,12 @@ private enum SelectionType {
                     }
                 } else {
                     if (Button(guiPos, nodeButtonStyle)) {
+                        Undo.RecordObject(target, $"Selected node {nIdx} of {spline.name}");
                         selection = n;
+                        selectionIndex = nIdx;
+                        spline.selectedNodeIndex = nIdx;
                         selectionType = SelectionType.Node;
+                        EditorUtility.SetDirty(target);
```

Comment:
Note that this is needed because GUI.changed will not become true when a GUI.Button is pressed.